pkg/sensors/tracing/kprobe_test.go

```diff
@@ -6142,6 +6142,60 @@ spec:
 	err = jsonchecker.JsonTestCheck(t, checker)
 	assert.NoError(t, err)
 }
+func TestKprobeUserStackTrace(t *testing.T) {
+	var doneWG, readyWG sync.WaitGroup
+	defer doneWG.Wait()
+
+	ctx, cancel := context.WithTimeout(context.Background(), tus.Conf().CmdWaitTime)
+	defer cancel()
+	testGetCpu := testutils.RepoRootPath("contrib/tester-progs/getcpu")
+	tracingPolicy := `apiVersion: cilium.io/v1alpha1
+kind: TracingPolicy
+metadata:
+  name: "test-user-stacktrace"
+spec:
+  kprobes:
+  - call: "sys_getcpu"
+    selectors:
+    - matchBinaries:
+      - operator: "In"
+        values:
+        - "` + testGetCpu + `"
+      matchActions:
+      - action: Post
+        userStackTrace: true`
+
+	createCrdFile(t, tracingPolicy)
+
+	obs, err := observertesthelper.GetDefaultObserverWithFile(t, ctx, testConfigFile, tus.Conf().TetragonLib, observertesthelper.WithMyPid())
+	if err != nil {
+		t.Fatalf("GetDefaultObserverWithFile error: %s", err)
+	}
+	observertesthelper.LoopEvents(ctx, t, &doneWG, &readyWG, obs)
+	readyWG.Wait()
+
+	if err := exec.Command(testGetCpu).Run(); err != nil {
+		t.Fatalf("failed to run %s: %s", testGetCpu, err)
+	}
+
+	stackTraceChecker := ec.NewProcessKprobeChecker("user-stack-trace").
+		WithProcess(ec.NewProcessChecker().WithBinary(sm.Full(testGetCpu))).
+		WithUserStackTrace(ec.NewStackTraceEntryListMatcher().WithValues(
+			ec.NewStackTraceEntryChecker().WithSymbol(sm.Suffix(("main.main"))),
```

Comment:
especially this version of the test will not pass 5.10. that would be better that each commit are separated logically and that they compile/pass the tests.